aws/resource_aws_iam_user_group_membership_test.go

```diff
@@ -0,0 +1,331 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/iam"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSUserGroupMembership_basic(t *testing.T) {
+	rString := acctest.RandString(8)
+	userName1 := fmt.Sprintf("tf-acc-ugm-basic-user1-%s", rString)
+	userName2 := fmt.Sprintf("tf-acc-ugm-basic-user2-%s", rString)
+	groupName1 := fmt.Sprintf("tf-acc-ugm-basic-group1-%s", rString)
+	groupName2 := fmt.Sprintf("tf-acc-ugm-basic-group2-%s", rString)
+	groupName3 := fmt.Sprintf("tf-acc-ugm-basic-group3-%s", rString)
+	membershipName := fmt.Sprintf("tf-acc-ugm-basic-membership-%s", rString)
+
+	usersAndGroupsConfig := testAccAWSUserGroupMembershipConfigUsersAndGroups(userName1, userName2, groupName1, groupName2, groupName3)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccAWSUserGroupMembershipDestroy,
+		Steps: []resource.TestStep{
+			// simplest test
+			resource.TestStep{
+				Config: usersAndGroupsConfig + testAccAWSUserGroupMembershipConfigInit(membershipName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aws_iam_user_group_membership.user1_test1", "user", userName1),
+					testAccAWSUserGroupMembershipCheckGroupListForUser(userName1, []string{groupName1}, []string{groupName2, groupName3}),
```

Comment:
It'd probably be good to also test that `userName2` is not attached to anything 👍